### PR TITLE
Add stable org.gnome.Contacts flatpak.

### PR DESCRIPTION
**Pull request by nielsdg**

### CODE REVIEW

**Comment by jurf** [comment]
bot, build org.gnome.Contacts

File: org.gnome.Contacts.json [hunk 1]
```diff
@@ -0,0 +1,215 @@
+{
+    "app-id": "org.gnome.Contacts",
+    "runtime": "org.gnome.Platform",
+    "runtime-version": "3.28",
+    "sdk": "org.gnome.Sdk",
+    "command": "gnome-contacts",
+    "tags": [ "stable" ],
```
Comment (TingPing) [inline org.gnome.Contacts.json]: Can remove this tag.

File: org.gnome.Contacts.json [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+    "rename-icon": "gnome-contacts",
```
Comment (TingPing) [inline org.gnome.Contacts.json]: Do you plan on renaming these upstream?

File: org.gnome.Contacts.json [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+        "--device=dri",
+        "--talk-name=org.freedesktop.Notifications",
+        "--talk-name=org.gnome.ControlCenter",
+        "--talk-name=org.gnome.OnlineAccounts",
+        "--talk-name=org.gnome.evolution.dataserver.AddressBook9",
+        "--talk-name=org.gnome.evolution.dataserver.Sources5",
+        "--talk-name=org.gnome.evolution.dataserver.Subprocess.Backend.*",
+        "--talk-name=org.freedesktop.Telepathy.AccountManager",
+        "--filesystem=home:ro",
+        "--filesystem=xdg-data/pixmaps/faces:ro",
```
Comment (TingPing) [inline org.gnome.Contacts.json]: Needs `:create` otherwise it will fail when it doesn't exist.

File: org.gnome.Contacts.json [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+        "--socket=wayland",
+        "--device=dri",
+        "--talk-name=org.freedesktop.Notifications",
+        "--talk-name=org.gnome.ControlCenter",
+        "--talk-name=org.gnome.OnlineAccounts",
+        "--talk-name=org.gnome.evolution.dataserver.AddressBook9",
+        "--talk-name=org.gnome.evolution.dataserver.Sources5",
+        "--talk-name=org.gnome.evolution.dataserver.Subprocess.Backend.*",
+        "--talk-name=org.freedesktop.Telepathy.AccountManager",
+        "--filesystem=home:ro",
```
Comment (TingPing) [inline org.gnome.Contacts.json]: What does it need home access for?

File: org.gnome.Contacts.json [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+                    "tag": "3.28.0",
+                    "commit": "3236929b648162c8a8b8c0020377f6d9af1ef6bb"
+                }
+            ]
+        },
+        {
+            "name": "libical",
+            "cleanup": [ "/lib/cmake"],
+            "buildsystem": "cmake",
+            "config-opts": [ "-DCMAKE_INSTALL_LIBDIR:PATH=/app/lib",
```
Comment (TingPing) [inline org.gnome.Contacts.json]: Don't need to specify libdir manually.

File: org.gnome.Contacts.json [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+        },
+        {
+            "name": "evolution-data-server",
+            "cleanup": [
+                "/lib/cmake",
+                "/lib/evolution-data-server/*-backends",
+                "/libexec",
+                "/share/dbus-1/services"
+            ],
+            "config-opts": [
```
Comment (TingPing) [inline org.gnome.Contacts.json]: Define CMAKE_BUILD_TYPE

File: org.gnome.Contacts.json [hunk 7]
```diff
@@ -0,0 +1,215 @@
...
+                    "type": "git",
+                    "url": "https://git.gnome.org/browse/cheese",
+                    "tag": "3.28.0",
+                    "commit": "355c121cd032837cad1bd54e1a560afc0712acba"
+                }
+            ]
+        },
+        {
+            "name": "gnome-contacts",
+            "buildsystem": "meson",
```
Comment (TingPing) [inline org.gnome.Contacts.json]: Add config-opts for `--buildtype=release` (or debugoptimized if you prefer)

File: org.gnome.Contacts.json [hunk 8]
```diff
@@ -0,0 +1,215 @@
...
+        {
+            "name": "libgdata",
+            "config-opts": [
+                "--disable-always-build-tests",
+                "--disable-Werror",
+                "--disable-static"
+            ],
+            "sources": [
+                {
+                    "type": "git",
```
Comment (TingPing) [inline org.gnome.Contacts.json]: It is a minor nit-pick but you will have noticeably faster build times, especially on the build bot, if you use tarball releases, at the very least for projects that don't update often.

File: org.gnome.Contacts.json [hunk 9]
```diff
@@ -0,0 +1,215 @@
...
+                "-DENABLE_WITH_PHONENUMBER=ON",
+                "-DENABLE_VALA_BINDINGS=ON",
+                "-DENABLE_WEATHER=OFF",
+                "-DWITH_OPENLDAP=OFF",
+                "-DENABLE_INTROSPECTION=ON",
+                "-DENABLE_INSTALLED_TESTS=OFF",
+                "-DENABLE_GTK_DOC=OFF",
+                "-DENABLE_EXAMPLES=OFF"
+            ],
+            "buildsystem": "cmake",
```
Comment (TingPing) [inline org.gnome.Contacts.json]: `cmake-ninja`

**Comment by TingPing** [comment]
https://flathub.org/builds/#/builders/2/builds/543

File: org.gnome.Contacts.json [hunk 10]
```diff
@@ -0,0 +1,215 @@
...
+            ]
+        },
+        {
+            "name": "gnome-contacts",
+            "buildsystem": "meson",
+            "sources": [
+                {
+                    "type": "git",
+                    "url": "https://gitlab.gnome.org/GNOME/gnome-contacts.git",
+                    "branch": "gnome-3-28"
```
Comment (jurf) [inline org.gnome.Contacts.json]: A Git tag + commit is preferred here I think.

File: org.gnome.Contacts.json [hunk 11]
```diff
@@ -0,0 +1,215 @@
...
+        "--talk-name=org.gnome.evolution.dataserver.AddressBook9",
+        "--talk-name=org.gnome.evolution.dataserver.Sources5",
+        "--talk-name=org.gnome.evolution.dataserver.Subprocess.Backend.*",
+        "--talk-name=org.freedesktop.Telepathy.AccountManager",
+        "--filesystem=home:ro",
+        "--filesystem=xdg-data/pixmaps/faces:ro",
+        "--filesystem=xdg-run/dconf", "--filesystem=~/.config/dconf:ro",
+        "--talk-name=ca.desrt.dconf", "--env=DCONF_USER_CONFIG_DIR=.config/dconf"
+    ],
+    "build-options" : {
```
Comment (jurf) [inline org.gnome.Contacts.json]: @TingPing: question: are these flags actually useful for anything here for example?

Comment (TingPing) [reply org.gnome.Contacts.json]: Some of the autotools systems *may* need them, its hard to say without reading build logs.

**Comment by barthalion** [comment]
Build failed for i386: https://flathub.org/builds/#/builders/22/builds/63/steps/5/logs/stdio

**Comment by jurf** [comment]
@barthalion just a rogue builder error, a rebuild built fine: https://flathub.org/builds/#/builders/2/builds/555

File: org.gnome.Contacts.json (hunk 4 above)
Comment (nielsdg) [reply org.gnome.Contacts.json]: To allow selecting an avatar image from a file.

File: org.gnome.Contacts.json (hunk 6 above)
Comment (nielsdg) [reply org.gnome.Contacts.json]: This should be solved by the line below, but Github doesn't seem to recognize it.

File: org.gnome.Contacts.json (hunk 10 above)
Comment (nielsdg) [reply org.gnome.Contacts.json]: Yes, going to release 3.28.1 now, so I'm immediately going to update it to that at the same time. :

**Comment by nedrichards** [comment]
bot, build org.gnome.Contacts

File: org.gnome.Contacts.json (hunk 7 above)
Comment (nielsdg) [reply org.gnome.Contacts.json]: Added!

**Comment by nielsdg** [comment]
bot, build org.gnome.Contacts

**Comment by nedrichards** [comment]
bot, build org.gnome.Contacts

File: org.gnome.Contacts.json (hunk 4 above)
Comment (nedrichards) [reply org.gnome.Contacts.json]: If you switch https://github.com/GNOME/gnome-contacts/blob/b4822d1040af1ff5858fe22924acbfb75a96ca7d/src/contacts-avatar-selector.vala#L225 for [FileChooserNative](https://developer.gnome.org/gtk3/stable/gtk3-GtkFileChooserNative.html) then you can drop this as it'll use the portal. I've filed this upstream as https://gitlab.gnome.org/GNOME/gnome-contacts/issues/96 it won't block this PR, but it'd be great to see it fixed in a future release.

**Comment by nedrichards** [comment]
This built fine (transient issue on i386 aside)

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/org.gnome.Contacts

**Comment by nielsdg** [comment]
Yippie! Thanks everyone!

**Pull request closed without merging**